src/smt/theory_str_noodler/theory_str_noodler_final_check.cpp

```diff
@@ -294,6 +328,10 @@ namespace smt::noodler {
 
                 if(m.is_false(block_len)) {
                     block_curr_len(block_len, false, true);
+                // if there are no length vars comming from the initial formula (or from axiom saturation),
+                // we can block the string assignment only
+                } else if(init_length_sensitive_vars.size() == 0) {
+                    block_curr_len(expr_ref(m.mk_false(), m));
```

Comment:
Why?

Reply:
If there are no lengths in the system, it should be sufficient to skip the generation of length formulae, no?

Reply:
But shouldn't the condition be the same as the one at the beginning of `dec_proc->get_lengths()`? We have no length vars AND no notcontains AND no disequations? Or something like that?

Reply:
Yes, I think you are right. I change it to take into account length variables from the procedure as they have to contain variables from disequations and notcontains for the option `ca_constr`.

Reply:
The code was modified to check for the size of `init_length_sensitive_vars`, so I assume we can close this now, or? @vhavlena

Reply:
ready, from my side

Reply:
I still do not understand what changed? You just check now for the initial vars from decision procedure, how do they differ from `init_length_sensitive_vars`?

Reply:
If we use notcontains/disequations handling using tag automata, it adds all variables among the length variables. If there is no length variable, there is no notcontains/disequations.

Reply:
Ah ok, makes sense. Just add this info into a comment, otherwise it looks very weird.

Reply:
Sure, done.